internal/venice-common/src/main/java/com/linkedin/venice/utils/Utils.java

```diff
@@ -306,18 +308,21 @@ public static boolean parseBooleanFromString(String value, String fieldName) {
   }
 
   /**
-   * For String-String key-value map config, we expect the command-line interface users to use "key1=value1,key2=value2,..."
+   * For String-String key-value map config, we expect the command-line interface users to use JSON
    * format to represent it. This method deserialize it to String-String map.
    */
-  public static Map<String, String> parseCommaSeparatedStringMapFromString(String value, String fieldName) {
+  public static Map<String, String> parseJsonMapFromString(String value, String fieldName) {
     try {
       Map<String, String> map = new HashMap<>();
       if (!value.isEmpty()) {
-        Arrays.stream(value.split(",")).map(s -> s.split("=")).forEach(strings -> map.put(strings[0], strings[1]));
+        ObjectMapper objectMapper = ObjectMapperFactory.getInstance();
+        return objectMapper.readValue(value, new TypeReference<Map<String, String>>() {
+        });
       }
       return map;
-    } catch (Exception e) {
-      throw new VeniceException(fieldName + " must be key value pairs separated by comma, but value: " + value);
+    } catch (IOException jsonException) {
+      throw new VeniceException(
+          fieldName + " must be a valid JSON object or key-value pairs separated by comma, but value: " + value);
```

Comment:
Let's edit this error message now that we don't support comma-separated key-value pairs